src/lib/marlin_plonk_bindings/js/bindings.js

```diff
@@ -463,3 +463,276 @@ var caml_pasta_fq_vector_emplace_back = function (v, x) {
+    return res;
+};
+
+// Provides: caml_pasta_pallas_sub
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_sub = function(x, y) {
+    var res = plonk_wasm.caml_pasta_pallas_sub(x, y);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_negate
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_negate = function(x) {
+    var res = plonk_wasm.caml_pasta_pallas_negate(x);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_double
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_double = function(x) {
+    var res = plonk_wasm.caml_pasta_pallas_double(x);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_scale
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_scale = function(x, y) {
+    var res = plonk_wasm.caml_pasta_pallas_scale(x, y);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_random
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_random = function() {
+    var res = plonk_wasm.caml_pasta_pallas_random();
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_rng
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_rng = function(i) {
+    var res = plonk_wasm.caml_pasta_pallas_rng(i);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_to_affine
+// Requires: plonk_wasm, rust_affine_to_caml_affine
+var caml_pasta_pallas_to_affine = function(pt) {
+    var res = plonk_wasm.caml_pasta_pallas_to_affine(pt);
+    return rust_affine_to_caml_affine(res);
+};
+
+// Provides: caml_pasta_pallas_of_affine
+// Requires: plonk_wasm, rust_affine_of_caml_affine, free_on_finalize
+var caml_pasta_pallas_of_affine = function(pt) {
+    var res = plonk_wasm.caml_pasta_pallas_of_affine(rust_affine_of_caml_affine(pt, plonk_wasm.caml_pasta_pallas_affine_one));
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_of_affine_coordinates
+// Requires: plonk_wasm, free_on_finalize
+var caml_pasta_pallas_of_affine_coordinates = function(x, y) {
+    var res = plonk_wasm.caml_pasta_pallas_of_affine_coordinates(x, y);
+    free_on_finalize(res);
+    return res;
+};
+
+// Provides: caml_pasta_pallas_endo_base
+// Requires: plonk_wasm
+var caml_pasta_pallas_endo_base = plonk_wasm.caml_pasta_pallas_endo_base;
+
+// Provides: caml_pasta_pallas_endo_scalar
+// Requires: plonk_wasm
+var caml_pasta_pallas_endo_scalar = plonk_wasm.caml_pasta_pallas_endo_scalar;
+
+// Provides: caml_pasta_pallas_affine_deep_copy
+// Requires: plonk_wasm, rust_affine_of_caml_affine, rust_affine_to_caml_affine
+var caml_pasta_pallas_affine_deep_copy = function(pt) {
+    return rust_affine_to_caml_affine(plonk_wasm.caml_pasta_pallas_affine_deep_copy(rust_affine_of_caml_affine(pt, plonk_wasm.caml_pasta_pallas_affine_one)));
+};
+
+
+
+
+
+// Provides: caml_pasta_vesta_one
```

Comment:
Idk if it's desirable, but you could use code to write all these functions for you like
```javascript
var list_of_function_names = [...];
list_of_function_namesforEach((name) => {
  function f() {
    var res = plonk_wasm[name].call(arguments);
    free_on_finalize(res);
    return res;
  }
  joo_global_object[name] = f;
});
```

Reply:
`js_of_ocaml` needs to be able to find the names as variables, so that it can inline calls and remove dead code. I think a helper might make things less boilerplatey though.